custodial-copy-reconciler/src/test/scala/uk/gov/nationalarchives/reconciler/DatabaseSpec.scala

```diff
@@ -0,0 +1,305 @@
+
+    additionalRowResponse.length should equal(1)
+    additionalRowResponse.head.representationType should equal("Preservation_1")
+    additionalRowResponse.head.coRef should equal(coRef2)
+
+    response.length should equal(1)
+    response.head.representationType should equal("Access_1")
+    response.head.coRef should equal(coRef)
+  }
+
+  "writeToActuallyInPsTable" should "given a coRef, delete the row and write a new one with the updated values to the ActualCosInPS table" in {
+    createActuallyInPsTable()
+    val ioRef = UUID.randomUUID()
+    val coRef = UUID.randomUUID()
+    val coRef2 = UUID.randomUUID()
+
+    val updatedCoRow = List(createPSCoRow(coRef, ioRef).unsafeRunSync().copy(sha256Checksum = Some("sha256Checksum1")))
+    val expectedAdditionalCoRow = List(createPSCoRow(coRef2, ioRef).unsafeRunSync())
+
+    Database[IO].writeToActuallyInPsTable(updatedCoRow).unsafeRunSync()
+    val response = getPreservicaCoRows(coRef).unsafeRunSync()
+    val additionalRowResponse = getPreservicaCoRows(coRef2).unsafeRunSync()
+
+    additionalRowResponse.length should equal(1)
+    additionalRowResponse.head.sha256Checksum should equal(None)
+    additionalRowResponse.head.coRef should equal(coRef2)
+
+    response.length should equal(1)
+    response.head.coRef should equal(coRef)
+    response.head.sha256Checksum should equal(Some("sha256Checksum1"))
+  }
+
+  "writeToExpectedInPsTable" should "should write nothing to the ExpectedInPs table if no CoRows were passed in" in {
+    createExpectedInPsTable()
+    val initialResponse = getOcflCoRows(coRef).unsafeRunSync()
+
+    Database[IO].writeToExpectedInPsTable(Nil).unsafeRunSync()
+    val response = getOcflCoRows(coRef).unsafeRunSync()
+
+    initialResponse should equal(Nil)
+    response should equal(Nil)
+  }
+
+  "writeToActuallyInPsTable" should "should write nothing to the ActualCosInPS table if no CoRows were passed in" in {
+    createActuallyInPsTable()
+    val initialResponse = getPreservicaCoRows(coRef).unsafeRunSync()
+
+    Database[IO].writeToActuallyInPsTable(Nil).unsafeRunSync()
+    val response = getPreservicaCoRows(coRef).unsafeRunSync()
+
+    initialResponse should equal(Nil)
+    response should equal(Nil)
+  }
+
+  "writeToExpectedInPsTable" should "return an error if there is an error with the ExpectedInPs table or DB" in {
+    val ex = intercept[Exception](Database[IO].writeToExpectedInPsTable(Nil).unsafeRunSync())
+    ex.getMessage should equal("[SQLITE_ERROR] SQL error or missing database (no such table: ExpectedCosInPS)")
+  }
+
+  "writeToActuallyInPsTable" should "return an error if there is an error with the ActualCosInPS table or DB" in {
+    val ex = intercept[Exception](Database[IO].writeToActuallyInPsTable(Nil).unsafeRunSync())
+    ex.getMessage should equal("[SQLITE_ERROR] SQL error or missing database (no such table: ActualCosInPS)")
+  }
+
+  val checksumMatchPossibilities
```

Comment:
This can be simplified a lot. We only care about sha256 as that's all the OCFL repo stores. 
```scala
  val checksumMatchPossibilities: TableFor3[Option[String], Option[String], Boolean] = Table(
    ("ocflChecksum", "preservicaChecksum", "matches"),
    ("checksum1".some, "checksum1".some, true),
    ("checksum1".some, "checksum2".some, false),
    ("checksum2".some, "checksum1".some, false),
    (None, "checksum1".some, false),
    ("checksum1".some, None, false),
    (None, None, false)
  )
```